flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/parser/SqlClientSyntaxHighlighter.java

```diff
@@ -0,0 +1,257 @@
+                } else {
+                    handleWord(word, highlightedOutput, null, style, true);
+                    currentParseState.getStyleSetter().accept(highlightedOutput, style);
+                    highlightedOutput.append(currentParseState.getStart());
+                    i += currentParseState.getStart().length() - 1;
+                }
+            } else {
+                word.append(currentChar);
+                final String stateEnd = currentParseState.getEnd();
+                if (buffer.regionMatches(i, stateEnd, 0, stateEnd.length())) {
+                    handleWord(word, highlightedOutput, currentParseState, style, true);
+                    i += stateEnd.length() - 1;
+                    currentParseState = null;
+                }
+            }
+        }
+        handleWord(word, highlightedOutput, currentParseState, style, false);
+        return highlightedOutput.toAttributedString();
+    }
+
+    private static void handleWord(
+            StringBuilder word,
+            AttributedStringBuilder sb,
+            State currentState,
+            SyntaxHighlightStyle style,
+            boolean turnOffHighlight) {
+        final String wordStr = word.toString();
+        if (currentState == null) {
+            if (flinkKeywordSet.contains(wordStr.toUpperCase(Locale.ROOT))) {
+                sb.style(style.getKeywordStyle());
+            } else {
+                sb.style(style.getDefaultStyle());
+            }
+            sb.append(wordStr);
+        } else if (turnOffHighlight) {
+            sb.append(wordStr);
+            final String stateEnd = currentState.getEnd();
+            if (stateEnd.length() > 1) {
+                sb.append(stateEnd.substring(1));
+            }
+        } else {
+            sb.append(wordStr);
+        }
+        word.setLength(0);
+        sb.style(style.getDefaultStyle());
+    }
+
+    /**
+     * State of parser while preparing highlighted output. This class represents a state machine.
+     *
+     * <pre>
+     *      MultiLine Comment                Single Line Comment
+     *           |                                    |
+     *   (&#47;*,*&#47;) |                                    | (--, \n)
+     *           *------------Default-----------------*
+     *                        |    |
+     *                        |    |
+     *           *------------*    *------------------*
+     *  (&#47;*+,*&#47;) |            |    |                  | (', ')
+     *           |            |    |                  |
+     *         Hint           |    |                String
+     *                        |    |
+     *                        |    |
+     *           *------------*    *------------------*
+     *    (&quot;, &quot;) |                                    | (`, `)
+     *           |                                    |
+     *     Hive Identifier                  Flink Default Identifier
+     * </pre>
+     */
+    private enum State {
```

Comment:
It's better we can add a State named `Default` to align with the java doc.

Reply:
added DEFAULT state as well